internal/merkle/proof_test.go

```diff
@@ -247,3 +971,86 @@ func (s *ComputeProofSuite) TestItMatchesMachineImplementation() {
 		root,
 	)
 }
+
+func FuzzVerifyProofs(f *testing.F) {
+	f.Add(uint(10), uint(10), uint(5))
+	f.Fuzz(func(t *testing.T, height, leafCount, leafIdx uint) {
+		height = bound(height, 3, 20)
+		leafCount = bound(leafCount, 1, uint(math.Pow(2, float64(height))))
+		leafIdx = bound(leafIdx, 0, leafCount-1)
+		leaves := generateRandomLeaves(leafCount)
+		leavesCopy := make([]model.Hash, len(leaves))
+		copy(leavesCopy, leaves)
+
+		root, siblings, err := CreateProofs(leavesCopy, height)
+		if err != nil {
+			t.Error(err)
+		}
+
+		leafSiblings := siblings[leafIdx*height : leafIdx*height+height]
+		newRoot := rootFromSiblings(leaves[leafIdx], int(leafIdx), leafSiblings)
+		if root != newRoot {
+			t.Errorf("expected: %v, got: %v\n", root, newRoot)
+		}
+
+		newRoot = rootFromSiblings(model.Hash{}, int(leafIdx), leafSiblings)
+		if root == newRoot {
+			t.Errorf("expected root to be different when replacing the leaf")
+		}
+
+		leafSiblings[0] = crypto.Keccak256Hash([]byte("wrong_sibling"))
```

Comment:
You could select a random sibling as well.

Reply:
Done at 92c00d6ec1ebed0c913cffac657d70596da188d7.